ovs/src/test/java/org/dcsa/conformance/standards/ovs/OvsChecksTest.java

```diff
@@ -0,0 +1,478 @@
+import org.dcsa.conformance.standards.ovs.checks.OvsChecks;
+import org.dcsa.conformance.standards.ovs.party.OvsFilterParameter;
+import org.dcsa.conformance.standards.ovs.party.SuppliedScenarioParameters;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class OvsChecksTest {
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private JsonNode serviceNodes;
+
+  @BeforeEach
+  void setUp() {
+    JsonNode vesselSchedules = createServiceVesselSchedules("1234567", "Great Vessel");
+    serviceNodes = createServiceNodes("Great Lion Service", "GLS", "SR12345A", vesselSchedules);
+  }
+
+  @Test
+  void testResponseContentChecks_validResponse() {
+    Set<String> issues =
+        executeResponseChecks(Map.of(OvsFilterParameter.CARRIER_SERVICE_CODE, "GLS"), serviceNodes);
+    assertTrue(issues.isEmpty());
+  }
+
+  @Test
+  void testResponseContentChecks_withWrongAttributesValuesResponse() {
+    JsonNode jsonBody = JsonToolkit.templateFileToJsonNode(
+      "/messages/ovs-300-response-wrong-attribute-values.json",
+      Map.ofEntries());
+
+    Set<String> issues =
+      executeResponseChecks(Map.of(OvsFilterParameter.CARRIER_SERVICE_CODE, "GLS"), jsonBody);
+    assertFalse(issues.isEmpty());
+  }
+
+  @Test
+  void testResponseContentChecks_withWrongDateTimesResponse() {
+    JsonNode jsonBody = JsonToolkit.templateFileToJsonNode(
+      "/messages/ovs-300-response-wrong-date-times.json",
+      Map.ofEntries());
+
+    Set<String> issues =
+      executeResponseChecks(Map.of(OvsFilterParameter.START_DATE, "2027-07-19"), jsonBody);
+    assertTrue(issues.isEmpty());
+  }
+
+  @Test
+  void testResponseContentChecks_withWrongStructureResponse() {
+    JsonNode jsonBody = JsonToolkit.templateFileToJsonNode(
+      "/messages/ovs-300-response-wrong-structure.json",
+      Map.ofEntries());
+
+    Set<String> issues =
+      executeResponseChecks(Map.of(OvsFilterParameter.CARRIER_SERVICE_CODE, ""), jsonBody);
+    assertFalse(issues.isEmpty());
+  }
+
+  private Set<String> executeResponseChecks(
+      Map<OvsFilterParameter, String> ovsFilterParameterStringMap, JsonNode serviceNodes) {
+    UUID matchedId = UUID.randomUUID();
+    String standardVersion = "3.0.0";
+    Supplier<SuppliedScenarioParameters> sspSupplier =
+        () -> SuppliedScenarioParameters.fromMap(ovsFilterParameterStringMap);
+    Set<String> issues = new HashSet<>();
+
+    ActionCheck actionCheck =
+        OvsChecks.responseContentChecks(matchedId, standardVersion, sspSupplier);
+    ((JsonAttributeBasedCheck) actionCheck)
+        .getValidators()
+        .forEach(
+            validator -> {
+              issues.addAll(validator.validate(serviceNodes));
+            });
+    return issues;
```

Comment:
If the checks are exposed on package level, you can remove this `executeResponseChecks` method.

Reply:
U still need the executeReponseChecks because you have the execute the validators. Hence it still remains in the unit test